From: Leonardo Arena <rnalrd@alpinelinux.org>
Date: Tue, 29 Sep 2026 09:47:41 +0200
Subject: [PATCH] Make --enable-udev_sync not require a libudev pkg-config
 file to be present

device-mapper's build target no longer calls any libudev API after
udev-sync-without-libudev-queue.patch, so treat a missing libudev.pc
as optional instead of a fatal configure error.

See: https://gitlab.alpinelinux.org/alpine/aports/-/work_items/18416
---

diff --git a/configure.ac b/configure.ac
index d64a4075..1561f2b9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1205,13 +1205,15 @@
 	[AC_MSG_ERROR([--enable-udev_sync requires udev >= 143. (--with-udev=$with_udev)])])
 
 AS_IF([test "$UDEV_SYNC" = "yes"], [
-	PKG_CHECK_MODULES(UDEV, libudev >= 143, [UDEV_PC="libudev"])
-	UDEV_STATIC_LIBS=$("$PKG_CONFIG" --static --libs libudev)
+	PKG_CHECK_MODULES(UDEV, libudev >= 143, [UDEV_PC="libudev"], [UDEV_PC=""])
 	AC_DEFINE([UDEV_SYNC_SUPPORT], 1, [Define to 1 to enable synchronization with udev processing.])
 
-	AC_CHECK_LIB(udev, udev_device_get_is_initialized, AC_DEFINE([HAVE_LIBUDEV_UDEV_DEVICE_GET_IS_INITIALIZED], 1,
-	  [Define to 1 if udev_device_get_is_initialized is available.]))
-	LIBS=$ac_check_lib_save_LIBS
+	AS_IF([test -n "$UDEV_PC"], [
+		UDEV_STATIC_LIBS=$("$PKG_CONFIG" --static --libs libudev)
+		AC_CHECK_LIB(udev, udev_device_get_is_initialized, AC_DEFINE([HAVE_LIBUDEV_UDEV_DEVICE_GET_IS_INITIALIZED], 1,
+		  [Define to 1 if udev_device_get_is_initialized is available.]))
+		LIBS=$ac_check_lib_save_LIBS
+	])
 ])
 
 dnl -- Enable udev rules
